File: backend/app/schemas/favorite.py
import uuid
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.book import BookResponse


class FavoriteCreate(BaseModel):
    openlibrary_work_id: str = Field(..., min_length=1, max_length=50, description="OpenLibrary Work ID")
    title: str = Field(..., min_length=1, max_length=500)
    authors: List[str] = Field(default_factory=list)
    cover_url: Optional[str] = None
    first_publish_year: Optional[int] = None
    description: Optional[str] = None
    edition_count: int = Field(default=1, ge=1)
    subjects: List[str] = Field(default_factory=list)


class FavoriteResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    book_id: uuid.UUID
    book: BookResponse
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class FavoriteListResponse(BaseModel):
    items: List[FavoriteResponse]
    total: int

File: book-finder/src/types/favorite.ts
import { Book } from './book';

export interface FavoriteItem {
  id: string;
  user_id: string;
  book_id: string;
  book: {
    id: string;
    openlibrary_work_id: string;
    title: string;
    authors: string[];
    cover_url: string | null;
    first_publish_year: number | null;
    description: string | null;
    edition_count: number;
    subjects: string[];
  };
  created_at: string;
}

export interface FavoriteCreatePayload {
  openlibrary_work_id: string;
  title: string;
  authors: string[];
  cover_url?: string | null;
  first_publish_year?: number | null;
  description?: string | null;
  edition_count?: number;
  subjects?: string[];
}

export interface FavoriteListResponse {
  items: FavoriteItem[];
  total: number;
}
